operator/api/vectorized/v1alpha1/cluster_types.go

```diff
@@ -192,6 +198,39 @@ type ClusterSpec struct {
 	NodePools []NodePoolSpec `json:"nodePools,omitempty"`
 }
 
+// ClusterConfiguration holds values (or references to values) that should be used
+// to configure the cluster. Where the cluster schema defines a non-string type for a
+// given key, the corresponding values here should be string-encoded (according to yaml
+// rules)
+// TODO: this type should be lifted to a shared module rather than duplicated in the `redpanda` CRD definition.
+type ClusterConfiguration struct {
+	Values map[string]ClusterConfigValue[string] `json:"values,omitempty"`
+}
+
+// ClusterConfigValue [T] holds a value of type T. For maps of arbitrary keys, non-string types
+// are string-encoded according to YAML rules in order to preserve accuracy.
+// TODO: this type should be lifted to a shared module rather than duplicated in the `redpanda` CRD definition.
+type ClusterConfigValue[T any] struct {
+	// If the value is directly known, it can be embedded here.
+	Value *T `json:"value,omitempty"`
+	// If the value is supplied by an k8s object reference, coordinates are embedded here.
+	// For non-string target types, the string value fetched from the source will be treated as
+	// a value encoded according to YAML rules.
+	ObjectRef *corev1.ObjectReference `json:"objectRef,omitEmpty"`
+	// If the value is supplied by an external source, coordinates are embedded here.
+	// For non-string target types, the string value fetched from the source will be treated as
+	// a value encoded according to YAML rules.
+	ExternalSecretRef *struct {
```

Comment:
the original agreement was that this feature will pretty much be hidden, also I think we'll be using just secret name as a generic/platform agnostic pointer to what you want to read.

We can also emit this from the first implementation.

For reference, what I am going to get from controlplane is `${secrets.PASSWORD}`